### PR TITLE
Make identity check more robust 

### PR DESCRIPTION
**Pull request by AWare**
Reverts guardian/subscriptions-frontend#954

Not forgetting that identity returns a 404 for not found.

### CODE REVIEW

**Comment by AWare** [comment]
We shouldn't even parse the JSON. 
200 -> 1 or more users
404 -> 0 users
other -> errro
https://github.com/guardian/identity/blob/3d607ee78fa92495913d9a63203b9b7be3ad279e/identity-api/src/main/scala/com/gu/identity/api/servlet/UserServlet.scala

File: app/services/IdentityService.scala [hunk 1]
```diff
@@ -26,7 +26,15 @@ class IdentityService(identityApiClient: => IdentityApiClient) extends LazyLoggi
 
   def doesUserExist(email: String): Future[Boolean] =
     identityApiClient.userLookupByEmail(email).map { response =>
-      (response.json \ "user" \ "id").asOpt[String].isDefined
+      response.status match {
+        case status if status == Status.OK =>  true
```
Comment (johnduffell) [inline app/services/IdentityService.scala]: I may be wrong, but I think you can just say `case Status.OK =>` and it'll just check equality.  You only need to name it if you're using it somewhere.

**Comment by johnduffell** [comment]
looks good, only a small comment! 👍 hopefully we didn't miss any response codes - may be worth checking with an identity person just to be sure ( @markjamesbutler or @jfsoul )

**Comment by jfsoul** [comment]
I suppose you might get a 403, but that would be unexpected and the catch-all case seems appropriate for that.

**Comment by AWare** [comment]
@johnduffell thanks :)

**Comment by johnduffell** [comment]
👍 great, looks great, thanks! 🚀

**Pull request merged**